pkg/elle/txn/cycle.go

```diff
@@ -0,0 +1,80 @@
+package txn
+
+import "github.com/pingcap/tipocket/pkg/elle/core"
+
+type FilterExType = func(cycleCase CycleCase) bool
+
+type CycleAnomalySpecType struct {
+	// A set of relationships which must intersect with every edge in the cycle.
+	Rels map[core.Rel]struct{}
+
+	// A set of relationships which must intersect with the first edge in the cycle.
+	FirstRel core.Rel
```

Comment:
Why the FirstRel isn't a set?

Reply:
It should be a set in original data structure, but all the `first-rel` set only contains one element.